tests/test_db.py

```diff
@@ -578,6 +637,198 @@ def test_operations(self):
+            db.add_group(self.trxl, self.org, None, '    ')
+
+        with self.assertRaisesRegex(ValueError, GROUP_NAME_WHITESPACE_ERROR):
+            db.add_group(self.trxl, self.org, None, '\t')
+
+        with self.assertRaisesRegex(ValueError, GROUP_NAME_WHITESPACE_ERROR):
+            db.add_group(self.trxl, self.org, None, '  \t  ')
+
+        # Check if operations have not been generated after the failure
+        operations = Operation.objects.all()
+        self.assertEqual(len(operations), 0)
+
+    def test_integrity_error(self):
+        """Check whether groups with the same group name cannot be inserted"""
+
+        group_name = 'errororg'
+
+        with self.assertRaisesRegex(AlreadyExistsError, DUPLICATED_GROUP_ERROR):
+            db.add_group(self.trxl, self.org, None, group_name)
+            db.add_group(self.trxl, self.org, None, group_name)
+
+    def test_integrity(self):
+        """Check whether group with the same group name can be inserted in two diff organizations"""
+
+        org1 = Organization.objects.create(name='Example1')
+        org2 = Organization.objects.create(name='Example2')
+        group_name = 'suborg'
+
+        group1 = db.add_group(self.trxl, org1, None, group_name)
+        group2 = db.add_group(self.trxl, org2, None, group_name)
+
+        self.assertIsInstance(group1,Group)
+        self.assertEqual(group1.organization, org1)
+
+        self.assertIsInstance(group2,Group)
+        self.assertEqual(group2.organization, org2)
+
+    def test_operations(self):
+        """Check if the right operations are created when adding a group"""
+
+        timestamp = datetime_utcnow()
+        db.add_group(self.trxl, self.org, None, 'suborg')
+
+        transactions = Transaction.objects.filter(name='add_group')
+        trx = transactions[0]
+
+        operations = Operation.objects.filter(trx=trx)
+        self.assertEqual(len(operations), 1)
+
+        op1 = operations[0]
+        self.assertIsInstance(op1, Operation)
+        self.assertEqual(op1.op_type, Operation.OpType.ADD.value)
+        self.assertEqual(op1.entity_type, 'group')
+        self.assertEqual(op1.trx, trx)
+        self.assertEqual(op1.target, 'suborg')
+        self.assertGreater(op1.timestamp, timestamp)
+
+        op1_args = json.loads(op1.args)
+        self.assertEqual(len(op1_args), 3)
+        self.assertEqual(op1_args['organization'], self.orgname)
+        self.assertEqual(op1_args['group_name'], 'suborg')
+        self.assertEqual(op1_args['parent'], None)
+
+
+class TestDeleteGroup(TestCase):
+    """Unit tests for delete_group"""
+
+    def setUp(self):
+        """Load initial dataset"""
+
+        self.user = get_user_model().objects.create(username='test')
+        self.ctx = SortingHatContext(self.user)
+
+        self.trxl = TransactionsLog.open('delete_group', self.ctx)
+        self.org = Organization.objects.create(name='Example')
+
+    def test_delete_group(self):
+        """Check whether it deletes a group"""
```

Comment:
Leave a blank line between docstring and code.

Reply:
Done 👍